microcloud/service/lxd.go

```diff
@@ -499,6 +499,58 @@ func (s LXDService) GetCephInterfaces(ctx context.Context, bootstrap bool, peers
 	return candidates, nil
 }
 
+// ValidateCephInterfaces validates the given interfaces map against the given Ceph network subnet
+// and returns a map of peer name to interfaces that are in the subnet.
+func (s *LXDService) ValidateCephInterfaces(cephNetworkSubnetStr string, interfacesMap map[string][]CephDedicatedInterface) (map[string][][]string, error) {
+	_, subnet, err := net.ParseCIDR(cephNetworkSubnetStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid CIDR subnet: %v", err)
+	}
+
+	ones, bits := subnet.Mask.Size()
+	if bits-ones == 0 {
+		return nil, fmt.Errorf("invalid Ceph network subnet (must be a subnet with at least one address)")
+	}
+
+	data := make(map[string][][]string)
+	for peer, interfaces := range interfacesMap {
+		for _, iface := range interfaces {
+			for _, addr := range iface.Addresses {
+				ip := net.ParseIP(addr)
+				if ip == nil {
+					return nil, fmt.Errorf("invalid IP address: %v", addr)
+				}
+
+				if subnet.IP.To4() != nil {
+					if ip.To4() != nil && subnet.Contains(ip) {
```

Comment:
Can the predicate here be moved into the one above to remove a layer of nesting? E.g.
```go
if subnet.IP.To4() != nil && ip.To4() != nil && subnet.Contains(ip) {
    // do IPv4 stuff
} else if subnet.IP.To6() != nil && ip.To6() != nil && subnet.Contains(ip) {
    // do IPv6 stuff
}
```

Reply:
fixed